### PR TITLE
added blockedTriageGroupAlert and new action and reducer for recipients

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- Added a new alert to be displayed when a user is trying to message triage groups that have been blocked or are no longer associated with the account.

### [MHV-52373](https://jira.devops.va.gov/browse/MHV-52373) - [MHV-52374](https://jira.devops.va.gov/browse/MHV-52374) - [MHV-52376](https://jira.devops.va.gov/browse/MHV-52376) - [MHV-52377](https://jira.devops.va.gov/browse/MHV-52377) - [MHV-52378](https://jira.devops.va.gov/browse/MHV-52378) ###

> Triage Group Blocking User Flows
> 1- User is blocked from some triage groups, but has others remaining (Compose)
> No message needed
> 
> API does not send a blocked triage group name, therefore, we can’t include on the dropdown.
> *Need a change in the API if this user flow should be different—where user sees the TG in the dropdown, but once the TG is chosen the error is displayed.
> (User receives a letter to let them know of the restriction via snail mail. There is some lag time—so, this flow may be confusing.)
> Do they get a phone call?
> Dropdown—See Joels mock with [BLOCKED] Oleksii said this will work from the dev perspective.
> 
> 2- User is blocked from all triage groups, no active TG
> Same as above
> a. Only one assigned, but blocked from that one. The list would be blank.
> b. Multiple TG, and blocked from all
> 
> 3- User has not been added to any triage groups
> 
> Currently this message is displayed when accessing SM:
> 
> *Have the API throw an error
> 4- User is replying to a message and has been blocked from the TG.
> This message should be more direct. Tell them WHY, instead of sending them to the coordinator immediately. (Too passive)
> 
> 5- Draft message created before blocking occurred

## Testing done

- Old behavior: users would be able to compose messages and reply to conversations that they have been blocked from or not associated with and would not know they could not send messages until trying to send a message.
- New behavior: An alert will appear on a message or message thread telling the user that they cannot message the group because they are blocked or not associated, and they will not be able to select groups they are blocked from.
- unit and e2e tests for each scenario

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| No Association |Screenshot 2024-01-04 at 9 57 43 AM|Screenshot 2024-01-04 at 9 57 05 AM|
| No Association At All |Screenshot 2024-01-04 at 10 12 14 AM|Screenshot 2024-01-04 at 10 11 57 AM|
| No Association At All |Screenshot 2024-01-04 at 10 56 23 AM|Screenshot 2024-01-04 at 10 55 48 AM|
| Blocked from 1 Care Team |Screenshot 2024-01-04 at 11 05 34 AM|Screenshot 2024-01-04 at 11 05 17 AM|
| Blocked from 1 Care Team (Existing Draft) |Screenshot 2024-01-04 at 11 09 11 AM|Screenshot 2024-01-04 at 11 08 42 AM|
| Blocked from Multiple Care Teams |Screenshot 2024-01-04 at 11 12 56 AM|Screenshot 2024-01-04 at 11 12 31 AM|
| Blocked from 1 Facility |Screenshot 2024-01-04 at 11 21 10 AM|Screenshot 2024-01-04 at 11 19 58 AM|
| Blocked from Multiple Teams and/or Facilities |Screenshot 2024-01-04 at 11 31 27 AM|Screenshot 2024-01-04 at 11 30 58 AM|
| Blocked from All Associated Groups |Screenshot 2024-01-04 at 12 39 23 PM|Screenshot 2024-01-04 at 12 39 09 PM|
| Blocked from All Associated Groups (Existing Draft) |Screenshot 2024-01-04 at 12 42 29 PM|Screenshot 2024-01-04 at 12 41 57 PM|
| Desktop |        |       |

## What areas of the site does it impact?

- VA.gov - Secure Messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Hey Kyle, this is enormous effort. Pretty excited and looking forward to this being released soon. I have few suggestions and some corrections. Happy to jump on a huddle to review these

File: src/applications/mhv/secure-messaging/tests/e2e/triage-team-tests/secure-messaging-thread-no-association-at-all.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    //
+    // // TODO move these assertion up after alert text fixing
+    //
+    // cy.get(Locators.ALERTS.BLOCKED_GROUP)
+    //   .find('p')
+    //   .should('have.text', Alerts.NO_ASSOCIATION.PARAGRAPH);
+    //
+    // cy.get(Locators.ALERTS.BLOCKED_GROUP)
+    //   .find('a')
+    //   .should('have.text', Alerts.NO_ASSOCIATION.LINK);
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/e2e/triage-team-tests/secure-messaging-thread-no-association-at-all.cypress.spec.js:114]: @fazilqa @haniasma do you guys need this commented code? If not, it is probably better to clean it up

File: src/applications/mhv/secure-messaging/components/ComposeForm/ReplyDraftItem.jsx [hunk 2]
```diff
@@ -483,29 +492,54 @@ const ReplyDraftItem = props => {
...
+                    setAttachFileSuccess={setAttachFileSuccess}
+                  />
+
+                  <FileInput
+                    attachments={attachments}
+                    setAttachments={setAttachments}
+                    setAttachFileSuccess={setAttachFileSuccess}
+                  />
+                </section>
+              )}
```
[the remark this answers is not in the document]
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/components/ComposeForm/ReplyDraftItem.jsx:532]: I tried this change, but it didn't quite work, so I left it as is.

File: src/applications/mhv/secure-messaging/tests/fixtures/json-triage-mocks/triage-teams-all-blocked-mock.json [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      "id": 2343333,
+      "name": "AnTran_TGTest",
+      "stationNumber": "636",
+      "blockedStatus": true,
+      "preferredTeam": true,
+      "relationshipType": "PATIENT"
+    }
+  ],
+  "mockBlokedFacilities": ["662", "636"]
+}
```
[the remark this answers is not in the document]
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/tests/fixtures/json-triage-mocks/triage-teams-all-blocked-mock.json:93]: I removed the "data" property from all the mock data files as discussed.

File: src/applications/mhv/secure-messaging/components/shared/BlockedTriageGroupAlert.jsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      if (associatedTriageGroupsQty === 0) {
+        return;
+      }
+
+      if (
+        associatedTriageGroupsQty > 0 &&
+        associatedTriageGroupsQty === associatedBlockedTriageGroupsQty
+      ) {
+        setAlertTitleText(
+          "You can't send messages to your care teams right now",
```
[the remark this answers is not in the document]
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/components/shared/BlockedTriageGroupAlert.jsx]: All the text here has been replaced with constants.

File: src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-links-buttons-landing-page.cypress.spec.js [hunk 5]
```diff
@@ -17,13 +17,19 @@ describe('Secure Messaging Verify Links and Buttons Keyboard Nav', () => {
...
+      .should('have.focus');
+    cy.realPress('Tab');
+    cy.get('[data-testid="faq-accordion-item"]').each(el => {
+      cy.get(el).should('have.focus');
+      cy.realPress('Tab');
+    });
+    // cy.tabToElement('.welcome-message > :nth-child(4) > a').should(
+    //   'have.focus',
+    // );
+    // cy.tabToElement('va-back-to-top').should('have.focus');
```
[the remark this answers is not in the document]
Comment (fazilqa) [reply src/applications/mhv/secure-messaging/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-links-buttons-landing-page.cypress.spec.js:32]: fixed and cleaned up

File: src/applications/mhv/secure-messaging/tests/e2e/triage-team-tests/secure-messaging-thread-no-association-at-all.cypress.spec.js (hunk 1 above)
Comment (fazilqa) [reply src/applications/mhv/secure-messaging/tests/e2e/triage-team-tests/secure-messaging-thread-no-association-at-all.cypress.spec.js:114]: fixed and cleaned up

File: src/applications/mhv/secure-messaging/tests/fixtures/json-triage-mocks/triage-teams-all-blocked-mock.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (fazilqa) [reply src/applications/mhv/secure-messaging/tests/fixtures/json-triage-mocks/triage-teams-all-blocked-mock.json:93]: I would like to participate in your conversation if any. I'm also do not have a clear picture about blocking feature.

**Review by alex-morgun-1 — DISMISSED** [review]

**Comment by oleksii-morgun-va** [comment]
Great job addressing those concerns. Approved. @fazilqa may be having some pending commits to clean up commented code in e2e tests before merging this

**Review by alex-morgun-1 — DISMISSED** [review]

**Review by alex-morgun-1 — DISMISSED** [review]

**Review by alex-morgun-1 — APPROVED** [review]
Recent changes look good 👍🏻

**Pull request merged**